### PR TITLE
Re-initializing devices if underlying pvc has been changed.

### PR DESCRIPTION
**Pull request by tanmayja**

### CODE REVIEW

**Comment by tanmayja** [comment]
TODO: update go.mod with updated aerospike-kubernetes-operator

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

File: pkg/update_pod_status_util.go
[the remark this answers is not in the document]
Comment (tanmayja) [reply pkg/update_pod_status_util.go:1]: dirtyVolumes does not have any dependency on pvcUID, and for sake of keeping it similar with initializedVolumes, we need to make changes in controller code as well.

File: pkg/update_pod_status_util.go [hunk 1]
```diff
@@ -212,22 +212,36 @@ func runBlkdiscard(logger logr.Logger, cmd []string, wg *sync.WaitGroup, guard c
...
+	pvcUID string) (needInit bool, volumes []string) {
 	for idx := range initializedVolumes {
 		initVolInfo := strings.Split(initializedVolumes[idx], "@")
 		if initVolInfo[0] == volName {
-			if len(initVolInfo) < 2 || initVolInfo[1] == pvcUID {
-				return false
+			if len(initVolInfo) < 2 {
+				initializedVolumes = append(initializedVolumes, fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID))
+
+				return false, remove(initializedVolumes, initializedVolumes[idx])
```
Comment (abhishekdwivedi3060) [inline pkg/update_pod_status_util.go:223]: We can directly overwrite the index as we not initialising here
initializedVolumes[idx] = fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID)

**Review by abhishekdwivedi3060 — CHANGES_REQUESTED** [review]

File: pkg/update_pod_status_util.go
[the remark this answers is not in the document]
Comment (abhishekdwivedi3060) [reply pkg/update_pod_status_util.go:1]: Yeah, at this point `dirtyVolumes` UID is not useful for local-provisioned PVC as we can always cleanup dirtyVolumes. It was just for consistency and future ref

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Review by sud82 — APPROVED** [review]

**Pull request merged**